logger.js

```diff
@@ -80,15 +81,16 @@ function pinoLogger (opts, stream) {
       return
     }
 
-    log[level]({
-      [resKey]: this,
-      [responseTimeKey]: responseTime
-    }, successMessage(this))
+    if (this.shouldLogSuccess || (autoLoggingStatusCodeThreshold && this.statusCode >= autoLoggingStatusCodeThreshold)) {
```

Comment:
I don't think `this.shouldLogSuccess` is actually needed, can you please move it back?

Reply:
The main reason to have it run on every finish is to capture the error handling logic on line 72, without it errors on ignored paths aren't logged unless the request receives an `error` event (and from the tests it looks like the client must trigger that manually?).

The two new tests `auto logging with autoLogging set to true and path ignored and error` and `auto logging with autoLogging set to true and path ignored and custom status code threshold` try to capture this and will fail with the old logic.